evaluation/functions.py

```diff
@@ -55,61 +64,73 @@ def process_prompt(prompt: str):
         end_time = time.time()
         answer = response["choices"][0]["messages"][1]["content"]
         response_time = end_time - start_time
-        return answer, response_time, response
+        return answer, response_time
 
     except (KeyError, IndexError) as e:
         raise Exception(f"Error in parsing the API response.\n{e}")
 
 
-# Function to rate the answer
-def evaluate_response(prompt: str):
+def evaluate_response(prompt: str, answer: str, evaluation_type: EvaluationType):
     """
     Rate the quality of an AI-generated response based on the provided response and original user question.
 
     Parameters:
     - prompt (str): The original user question.
+    - answer (str): The AI-generated response to be evaluated.
+    - evaluation_type (EvaluationType): The type of evaluation to be performed.
     """
-    answer, response_time, response = process_prompt(prompt)
-    evaluation_string = os.environ.get(
-        "PROMPT_EVAL_STRING", "Evaluate the above response quality"
-    )
+    eval_prompts = {
+        EvaluationType.RELEVANCE: "judge the relevance of the output above to the given question and context and score the output on a scale of 1-5.",
+        EvaluationType.ACCURACY: "judge the accuracy of the output above to the given question and context and score the output on a scale of 1-5.",
+        EvaluationType.COHERENCE: "judge the coherence of the output above to the given question and context and score the output on a scale of 1-5.",
+        EvaluationType.FLUENCY: "judge the fluency of the output above to the given question and context and score the output on a scale of 1-5.",
+        EvaluationType.GROUNDEDNESS: "judge the groundedness of the output above to the given question and context and score the output on a scale of 1-5.",
+    }
+
     try:
         prompt_data = {
             "messages": [
                 {
                     "role": "user",
-                    "content": prompt,
-                },
-                {
-                    "role": "tool",
-                    "content": response["choices"][0]["messages"][0]["content"],
-                },
-                {
-                    "role": "assistant",
-                    "content": response["choices"][0]["messages"][1]["content"],
-                },
-                {"role": "user", "content": evaluation_string},
+                    "content": f"Question: {prompt}?\nResponse: {answer}\n{eval_prompts[evaluation_type]}",
+                }
             ]
         }
         response = get_response(prompt_data)
         evaluation = response["choices"][0]["messages"][1]["content"]
-        return prompt, answer, response_time, evaluation
+        return evaluation
+
+    except (KeyError, IndexError) as e:
+        raise Exception(f"Error in parsing the API response.\n{e}")
+
 
+def paraphrase_question(prompt: str):
```

Comment:
What is the purpose for paraphrasing?

Reply:
The reason for the question is that the Azure OpenAI endpoints used by the main app (and by extension the eval scripts) already do query rewriting.